qgreenland/util/model_validators/layer_style.py

```diff
@@ -0,0 +1,115 @@
+    * Ensures continuous legends are horizontal.
+    * Ensures a "Suffix" is populated in the "Legend Settings" menu.
+    """
+    if not style_name:
+        return style_name
+
+    style_filepath = get_style_filepath(style_name)
+    tree = ElementTree.parse(style_filepath)
+
+    errors = []
+    error_prefix = (
+        f"Style '{style_name}' has a continuous legend that is incorrectly"
+        " configured."
+        f" In QGIS >=3.28, edit this style ({style_filepath}) in the layer symbology"
+        ' menu and configure the continuous legend in the "Legend Settings" submenu to'
+        " resolve the following:"
+    )
+
+    colorrampshader = tree.find(".//colorrampshader")
+    if colorrampshader is None:
+        # This style is not using a colorramp, so there is no "Label unit suffix"
+        # setting to be concerned with.
+        return style_name
+
+    if colorrampshader.attrib["colorRampType"] != "INTERPOLATED":
+        # This colorramp should not have a continuous legend, so the  "Label unit
+        # suffix" will be used if specified.
+        return style_name
+
+    rampLegendSettings = colorrampshader.find(".//rampLegendSettings")
+    if not rampLegendSettings:
+        raise exc.QgrInvalidConfigError(
+            f"{error_prefix}\n"
+            "  * Continuous (i.e. linearly interpolated) colorramps must be"
+            " re-configured with a newer version of QGIS to support continuous legends."
+            ' Please ensure that the unit of measurement is populated in the "Suffix"'
+            ' field, and that the "Orientation" field is set to "Horizontal".\n'
+        )
+
+    if (orientation := rampLegendSettings.attrib["orientation"]) != "1":
+        errors.append(
+            f'"Orientation" must be horizontal ("1"). Is currently "{orientation}"'
+        )
+
+    if not (suffix := rampLegendSettings.attrib["suffix"]):
+        # Populating " " is a workaround for the rare case a layer with a continuous
+        # legend has no unit of measurement. Validating the Suffix matches the
+        # "Layer unit suffix" may be a more thorough check, but it's much more
+        # difficult because QGIS doesn't store "Layer unit suffix" in any particular
+        # field. It's calculated on-the-fly in an unreliable way, depending on QGIS
+        # version.
+        errors.append(
+            f'"Suffix" must contain a unit of measurement. Is currently "{suffix}".'
+            " If this layer truly has no unit of measurement (e.g. NDVI or a count),"
+            ' populate a space (" ") to silence this error'
+        )
+
+    if errors:
+        # NOTE: chr(92) is a backslash to work around "f-string expression part
+        # cannot include a backslash" and still put a newline in there.
+        newline = "\n"
+        raise exc.QgrInvalidConfigError(
+            f"{error_prefix}\n" f"{newline.join(f'  * {err}' for err in errors)}\n"
```

Comment:
Maybe worth putting a link to the docs in this message? At thee end of the newline-joined list? It would look something like:

> See https://qgreenland.readthedocs.io/en/latest/contributor/how-to/contribute-styles.html#continuous-colormap-considerations

Reply:
Yes! I remember now you suggested that earlier. Thanks for the reminder :)

Reply:
ca1248dd